lib/util/http/host-rules.ts

```diff
@@ -162,6 +163,18 @@ export function applyHostRules<GotOptions extends HostRulesGotOptions>(
   if (!hasProxy() && foundRules.enableHttp2 === true) {
     options.http2 = true;
   }
+
+  if (
+    is.nonEmptyString(foundRules.certificateAuthority) ||
+    is.nonEmptyString(foundRules.privateKey) ||
+    is.nonEmptyString(foundRules.certificate)
+  ) {
+    options.https = {
+      certificateAuthority: foundRules.certificateAuthority,
+      key: foundRules.privateKey,
+      certificate: foundRules.certificate,
```

Comment:
they seem to pass them down as is.

https://github.com/sindresorhus/got/blob/main/source/core/options.ts#L2432-L2434

Reply:
Is that a problem or are they eventually ignored?

Reply:
I don't seem `got` doing anything special around those params. they are passed down to [`https.request`](https://nodejs.org/api/https.html#httpsrequesturl-options-callback) method as is. My guess is that null values are eventually ignored

Reply:
turned out my guess was wrong after some local test with the got library. got actually checks what's been set to the `https` object. I've made the change to separate the attribute assignment so that this won't happen. 

https://github.com/sindresorhus/got/blob/main/source/core/options.ts#L2212-L2214